engine/physics/src/physics/box2d/box2d_physics.cpp

```diff
@@ -404,7 +404,7 @@ namespace dmPhysics
                     polygon_shape_data->m_Polygon.vertices[i].x = p.x * s;
                     polygon_shape_data->m_Polygon.vertices[i].y = p.y * s;
                 }
-
+                polygon_shape_data->m_Polygon.centroid = polygon_shape_data->m_CentroidOriginal * s;
```

Comment:
In order to be able to scale shapes, we store the original vertices, and apply the scaling factor.
We also need to do the same for the centroid, or the centroid may end up _outside_ of the polygon, which isn't allowed.